src/main/java/org/broadinstitute/consent/http/health/SendGridHealthCheck.java

```diff
@@ -0,0 +1,68 @@
+package org.broadinstitute.consent.http.health;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.google.api.client.http.HttpStatusCodes;
+import com.google.gson.Gson;
+import io.dropwizard.lifecycle.Managed;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.broadinstitute.consent.http.configurations.MailConfiguration;
+import org.broadinstitute.consent.http.util.HttpClientUtil;
+
+import java.nio.charset.Charset;
+
+public class SendGridHealthCheck extends HealthCheck implements Managed {
+    private final HttpClientUtil clientUtil;
+    private final MailConfiguration configuration;
+
+    public SendGridHealthCheck(HttpClientUtil clientUtil, MailConfiguration configuration) {
+        this.clientUtil = clientUtil;
+        this.configuration = configuration;
+    }
+
+    @Override
+    protected Result check() throws Exception {
+        try {
+            String statusUrl = configuration.getSendGridStatusUrl();
+            HttpGet httpGet = new HttpGet(statusUrl);
+            try (CloseableHttpResponse response = clientUtil.getHttpResponse(httpGet)) {
+                if (response.getStatusLine().getStatusCode() == HttpStatusCodes.STATUS_CODE_OK) {
+                    String content = IOUtils.toString(response.getEntity().getContent(), Charset.defaultCharset());
+                    SendGridStatus sgStatus = new Gson().fromJson(content, SendGridStatus.class);
+                    if (sgStatus.status.description.equalsIgnoreCase("All Systems Operational")) {
+                        return Result.builder()
+                                .withDetail("pages", sgStatus.page)
+                                .withDetail("status", sgStatus.status)
```

Comment:
Try to use getters for more idiomatic Java. If the class doesn't have them, please add them. Direct field references are great in other languages, less so in Java.
```suggestion
                                .withDetail("pages", sgStatus.getPage())
                                .withDetail("status", sgStatus.getStatus())
```